src/pip/_internal/network/download.py

```diff
@@ -175,8 +163,15 @@ def __call__(self, link):
             )
             raise
 
-        return Download(
-            resp,
-            _get_http_response_filename(resp, link),
-            _prepare_download(resp, link, self._progress_bar),
-        )
+        filename = _get_http_response_filename(resp, link)
+        chunks = _prepare_download(resp, link, self._progress_bar)
+        with open(os.path.join(location, filename), 'wb') as content_file:
+            for chunk in chunks:
+                content_file.write(chunk)
+            return content_file.name, resp.headers.get('Content-Type', '')
```

Comment:
> Let's change the return type here, to be inline with what is returned as part of download_many's iterable, which simplifies call sites and download_many.

The return type of download_many is intended to be similar to that of enumerate, i.e. to keep track of which file belong to which link since probably we don't want to warranty it to retain the order.  Currently that method is only a placeholder, while the caller of download_one doesn't need the URL anyhow.  Do you still want me to change the return type here?

Reply:
Yes, that sounds fair. :)